hw/xbox/dsp/dsp_cpu.c

```diff
@@ -1019,7 +1021,7 @@ static void write_memory_raw(dsp_core_t* dsp, int space, uint32_t address, uint3
         dsp->yram[address] = value;
```

Comment:
What about here too?

Reply:
It looked to me like these areas are only accessed by the dsp code as 32bit words and so are not effected by endianness issues: since it is never written or read by anything else. (But do correct me if I’m wrong) 

pram, however, is binary code transferred from the cpu, so we need to respect endianness here.

Reply:
Generally P, X, Y - banks are all the same: 24 bit words.

The main means of accessing them with data is through MMIO regions (mapping directly to DSP RAM) or DMA (controlled by DSP code).

I'm not sure how the Hatari code accesses each of these banks. Our DSP DMA seems to use `dsp56k_read_memory` / `dsp56k_write_memory` to access each of these banks.

There's also scratch space, which isn't 24 bit words (and probably poorly accessed):

https://github.com/xqemu/xqemu/blob/787012e012af96d6d69ebac3f32a63356ede93c2/hw/xbox/dsp/dsp_dma.c#L271-L288